src/launchpad/size/insights/apple/image_optimization.py

```diff
@@ -36,17 +37,26 @@ class _OptimizationResult:
     optimized_size: int
 
 
-class ImageOptimizationInsight(Insight[ImageOptimizationInsightResult]):
-    """Analyse image optimisation opportunities in iOS apps."""
+class BaseImageOptimizationInsight(Insight[ImageOptimizationInsightResult], ABC):
+    """Base class for image optimization insights with shared analysis logic."""
 
     OPTIMIZABLE_FORMATS = {"png", "jpg", "jpeg", "heif", "heic"}
     MIN_SAVINGS_THRESHOLD = 4096
     TARGET_JPEG_QUALITY = 85
     TARGET_HEIC_QUALITY = 85
     _MAX_WORKERS = 4
 
+    @abstractmethod
+    def _iter_files_to_analyze(self, input: InsightsInput) -> Iterable[FileInfo]:
```

Comment:
Can these two methods be combined into something like `_find_images() -> List[FileInfo]`? The de-duping can be merged into that as well. The current structure here is a bit awkward IMO, the iterator just converts it to a list anyways. It looks like the subclasses are responsible for returning a list of images that should be considered for processing.

Reply:
Similarly, the `_analyze_image_optimization` override doesn't fit in nicely here since we lose the ThreadPoolExecutor logic and duplicate a lot of size-checking code. Maybe there could be some kind of overridable "preprocess" or "transform" method that gets called before `_analyze_image_optimization` so that method doesn't have to be overriden?